### PR TITLE
Weather api

### LINKED ISSUE #243: [DS] As a user, I want the app to integrate a weather forecast service, allowing me to view weather conditions and forecasts for upcoming matches or training sessions, enabling better planning and preparation based on the weather.
Here is how we "saw" this could be implemented. These are just ideas, the implementation details are up to you

- Based on the location of the event
- Well integrated in the app, not just a page with a random weather map fragment (instead for example, display a weather icon for each events in the search list, an overlay on google maps, in the event page itself). Weather constraints for sports could be either hard-coded (jogging outside obviously depends on weather) or specified by users?
- Alerts, updates on events if there is bad weather for everyone who has joined.

### PR DESCRIPTION
**Pull request by ziazi**
This pull request contains the necessary methods and classes to fetch (synchronously and asynchronously) the weather from visualcrossing given a timestamp and location.

### CODE REVIEW

File: app/src/main/java/com/github/versus/weather/WeatherService.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+import java.util.concurrent.CompletableFuture;
+
+import retrofit2.Call;
+import retrofit2.Response;
+import retrofit2.Retrofit;
+import retrofit2.converter.gson.GsonConverterFactory;
+
+public final class WeatherService {
+    private static final String BASE_URL = "https://weather.visualcrossing.com/";
+    private static final String API_KEY= "H4DNTYBMHPAU7EC3D4KC3W7KW";
```
Comment (hamzaremmal) [inline app/src/main/java/com/github/versus/weather/WeatherService.java:18]: You should remove the API_KEY from this file as it should not be visible.

Comment (ziazi) [reply app/src/main/java/com/github/versus/weather/WeatherService.java:18]: It's not a problem for now

**Review by hamzaremmal — DISMISSED** [review]
All this looks good but the visible API_KEY in WeatherService. You should revoke this key and generate a new one. Try to use local.properties file and environment variables (same approach as the Google Maps API key)

**Review by Adam-GitHub0 — COMMENTED** [review]
For the API Key , if you want you can use the same code I used for my Google Maps key :
```java
try {
            API_KEY = getContext().getPackageManager().getApplicationInfo(getContext().getPackageName(), 
             PackageManager.GET_META_DATA).metaData.getString("com.google.android.geo.API_KEY");
        
             } catch (PackageManager.NameNotFoundException e) {
            throw new RuntimeException(e);
        }
```
            
        You would need to change the argument of getString and I think it will work !

**Comment by ziazi** [comment]
> For the API Key , if you want you can use the same code I used for my Google Maps key :
> 
> ```java
> try {
>             API_KEY = getContext().getPackageManager().getApplicationInfo(getContext().getPackageName(), 
>              PackageManager.GET_META_DATA).metaData.getString("com.google.android.geo.API_KEY");
>         
>              } catch (PackageManager.NameNotFoundException e) {
>             throw new RuntimeException(e);
>         }
> ```
> 
>  ```
>     You would need to change the argument of getString and I think it will work !
> ```

Thanks for the tip! However it's not this that is bothering me.
I actually want later to have (next sprint) multiple API keys and switch between them when I get an http 429 since I have a limited (and too few in my opinion) number of queries. So my plan would be to have many of them and then they wouldn't be hardcoded. But for the sake of this sprint I'll just leave it like this because I regularly need to change keys and I need to worry about showing weather in posts now.

**Comment by Adam-GitHub0** [comment]
> > For the API Key , if you want you can use the same code I used for my Google Maps key :
> > ```java
> > try {
> >             API_KEY = getContext().getPackageManager().getApplicationInfo(getContext().getPackageName(), 
> >              PackageManager.GET_META_DATA).metaData.getString("com.google.android.geo.API_KEY");
> >         
> >              } catch (PackageManager.NameNotFoundException e) {
> >             throw new RuntimeException(e);
> >         }
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > ```
> >    You would need to change the argument of getString and I think it will work !
> > ```
> 
> Thanks for the tip! However it's not this that is bothering me. I actually want later to have (next sprint) multiple API keys and switch between them when I get an http 429 since I have a limited (and too few in my opinion) number of queries. So my plan would be to have many of them and then they wouldn't be hardcoded. But for the sake of this sprint I'll just leave it like this because I regularly need to change keys and I need to worry about showing weather in posts now.

Okay I see now  ! 
By the way , really good  work ! Keep it up !

**Review by Adam-GitHub0 — APPROVED** [review]

**Pull request merged**